unic-locale-impl/src/extensions/unicode.rs

```diff
@@ -63,10 +63,74 @@ fn is_type(t: &str) -> bool {
 }
 
 impl UnicodeExtensionList {
+    /// Returns `true` if there are no keywords and no attributes in
+    /// the `UnicodeExtensionList`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use unic_locale_impl::Locale;
+    ///
+    /// let mut lo: Locale = "en-US-u-foo".parse()
+    ///     .expect("Parsing failed.");
+    ///
+    /// assert_eq!(lo.extensions.unicode.is_empty(), false);
+    /// ```
     pub fn is_empty(&self) -> bool {
         self.keywords.is_empty() && self.attributes.is_empty()
     }
 
+    /// Returns the value of keyword in the `UnicodeExtensionList`.
+    ///
+    /// NB: value here is referred to as type in UTS #35.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use unic_locale_impl::Locale;
+    ///
+    /// let mut lo: Locale = "en-US-u-ca-buddhist".parse()
+    ///     .expect("Parsing failed.");
+    ///
+    /// assert_eq!(lo.extensions.unicode.get_keyword("ca")
+    ///                .expect("Getting keyword failed."),
+    ///            Some(vec!("buddhist")));
+    ///
+    /// // Here keyword with key "aa" is not available
+    /// assert_eq!(lo.extensions.unicode.get_keyword("aa")
+    ///                .expect("Getting keyword failed."),
+    ///            None);
+    /// ```
+    pub fn get_keyword(&self, key: &str) -> Result<Option<Vec<&str>>, LocaleError> {
```

Comment:
Same question about `Result<Option<impl ExactSizeIterator>>`, and `AsRef<[u8]>`.